runtime/queries/metricsview_aggregation.go

```diff
@@ -130,6 +132,48 @@ func (q *MetricsViewAggregation) Export(ctx context.Context, rt *runtime.Runtime
 	}
 	defer e.Close()
 
+	res, err := rt.Resolve(ctx, &runtime.ResolveOptions{
+		InstanceID: instanceID,
+		Resolver:   "metrics_time_range",
+		ResolverProperties: map[string]any{
+			"metrics_view": q.MetricsViewName,
+		},
+		Args: map[string]any{
+			"priority": q.Priority,
+		},
+		Claims: q.SecurityClaims,
+	})
+	if err != nil {
+		return err
+	}
+	defer res.Close()
+
+	row, err := res.Next()
+	if err != nil {
+		if errors.Is(err, io.EOF) {
+			return errors.New("time range query returned no results")
+		}
+		return err
+	}
+
+	minTime, err := anyToTime(row["min"].(string))
+	if err != nil {
+		return err
+	}
+	maxTime, err := anyToTime(row["max"].(string))
+	if err != nil {
+		return err
+	}
+	watermark, err := anyToTime(row["watermark"].(string))
+	if err != nil {
+		return err
+	}
+
+	err = e.BindQuery(ctx, qry, minTime, maxTime, watermark)
+	if err != nil {
+		return err
+	}
```

Comment:
The export can take a long time, so it would be best for `res.Close()` to run earlier than the end of the function. Perhaps pull this into a smaller scope or util function

Reply:
Moved to a util funciton